### PR TITLE
Adding unit test for list volume flow

### PR DESCRIPTION
**Pull request by sathishkumarsivagurunathan**
Adding list volume for unit test.

### CODE REVIEW

File: test/listvolume_tester.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        mock_3parclient = self.mock_objects['mock_3parclient']
+        mock_3parclient.getWsApiVersion.assert_called()
+
+    def get_request_params(self):
+        return {"Name": "test-vol-001",
+                "Opts": {}}
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
```
Comment (imran-ansari) [inline test/listvolume_tester.py]: Would be nice to return dummy list of volumes from mock_etcd.get_all_vols(). Also we could have two TCs - one for mounted volumes list and other for unmounted volumes list

File: test/listvolume_tester.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": '', 'Volumes':[]})
+
+        # Check if these functions were actually invoked
+        # in the flow or not
+        mock_3parclient = self.mock_objects['mock_3parclient']
+        mock_3parclient.getWsApiVersion.assert_called()
+
+    def get_request_params(self):
+        return {"Name": "test-vol-001",
```
Comment (imran-ansari) [inline test/listvolume_tester.py]: request params can be an empty dictionary as for list_volumes, we don't get any parameters.

File: test/listvolume_tester.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    def override_configuration(self, config):
+        pass
+
+    # TODO: check_response and setup_mock_objects can be implemented
+    # here for the normal happy path TCs here as they are same
+
+
+class TestListVolumeDefault(ListVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": '', 'Volumes':[]})
```
Comment (imran-ansari) [inline test/listvolume_tester.py]: Response will have list of volumes. So comparison can be made to what is received vs what was set to return from mock_etcd.get_all_vols()

File: test/listvolume_tester.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+        # Check if these functions were actually invoked
+        # in the flow or not
+        mock_3parclient = self.mock_objects['mock_3parclient']
+        mock_3parclient.getWsApiVersion.assert_called()
+
```
Comment (imran-ansari) [inline test/listvolume_tester.py]: There are going to be ETCD calls like get_all_vols() and get_path_info_from_vol() during the flow execution. Assert can be made to ensure that those calls were indeed called.

Comment (sathishkumarsivagurunathan) [reply test/listvolume_tester.py]: Done

File: test/listvolume_tester.py (hunk 2 above)
Comment (sathishkumarsivagurunathan) [reply test/listvolume_tester.py]: Done

File: test/listvolume_tester.py (hunk 1 above)
Comment (sathishkumarsivagurunathan) [reply test/listvolume_tester.py]: Tried it. But I dont have the idea what is the expected out for get_all_vols call.. This is making it fail.. Would be great if you can share an example response.

File: test/listvolume_tester.py (hunk 3 above)
Comment (sathishkumarsivagurunathan) [reply test/listvolume_tester.py]: Depends on the first mentioned comment. ["Tried it. But I dont have the idea what is the expected out for get_all_vols call.. This is making it fail.. Would be great if you can share an example response."]

**Comment by wdurairaj** [comment]
@imran-ansari ,  do you want this pull request to be still worked upon ?

**Comment by imran-ansari** [comment]
Yes, this needs to be worked upon.

**Comment by wdurairaj** [comment]
Will be handled by pull request -- https://github.com/hpe-storage/python-hpedockerplugin/pull/334

**Pull request closed without merging**